lib/src/client/query_parameter.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Enclosing class for query parameters that can accommodate a [List] of
```

Comment:
I don't think this class is needed if you apply changes suggested below.

Reply:
I'd still like to keep this class because we plan to eventually add a `List<QueryParameter>` to `WebCallOptions`. The reason to use this class instead of a simple `Map<String, String>` is because this class supports multi-value parameters.

For now though, the simplified code in `cors.dart` means the import isn't (for now) needed there anymore. The class is also not yet exposed in the exports in `grpc_web.dart`.

Reply:
The static typing is one reason, the sorting behavior is another. Ensuring that the list of query params is stable is helpful for testing as well as any client-side caching behavior that may be based on this.